File: BackEnd/ViewModels/Login/RegisterModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace sd2376_workshop_backend.ViewModels.Login
{
    public class RegisterModel
    {
        public int Id { get; set; }

        [Required]
        public string Name { get; set; }

        [Required]
        public string Password { get; set; }

        [Required]
        [EmailAddress(ErrorMessage = "Invalid email")]
        public string Email { get; set; }

        public bool IsActive { get; set; }

        public IList<int> RoleIds { get; set; } = new List<int>();
    }
}

File: FrontEnd/src/app/models/register.model.ts
export class RegisterModel {
    constructor( id?: number, name?:string,email?: string,password?: string, token?: string)
    {
        this.id = id || 0,
        this.name=name || '',
        this.email=email || '',
        this.password=password || '',        
        this.token=token || ''
    }
    public id: number;
    public name: string;
    public email: string ;
    public password: string;    
    public token: string;
}